src/strategies/BeefyPearlStrategy.sol

```diff
@@ -0,0 +1,274 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.16;
+
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {SafeTransferLib} from "solmate/src/utils/SafeTransferLib.sol";
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+
+import {IRouter, IPair} from "src/interfaces/IPearl.sol";
+import {IBeefyVault} from "src/interfaces/Beefy.sol";
+
+import {AccessStrategy} from "src/strategies/AccessStrategy.sol";
+
+import {AffineVault} from "src/vaults/Vault.sol";
+import {StrategyVault} from "src/vaults/locked/StrategyVault.sol";
+
+contract BeefyPearlStrategy is AccessStrategy {
+    using FixedPointMathLib for uint256;
+    using SafeTransferLib for ERC20;
+
+    // beefy vault
+    IBeefyVault public immutable beefy;
+    // const max bps
+    uint256 public constant MAX_BPS = 10_000;
+    uint256 public defaultSlippageBps;
+
+    ERC20 public immutable token0;
+    ERC20 public immutable token1;
+
+    IRouter public immutable pearlRouter;
+    IPair public immutable lpToken;
+
+    constructor(AffineVault _vault, IBeefyVault _beefy, IRouter _router, ERC20 _token1, address[] memory strategists)
+        AccessStrategy(_vault, strategists)
+    {
+        beefy = _beefy;
+        token0 = asset;
+        token1 = _token1;
+
+        pearlRouter = _router;
+        lpToken = IPair(pearlRouter.pairFor(address(token0), address(token1), true));
+
+        require(pearlRouter.isPair(address(lpToken)), "BPS: Invalid pearl LP pair");
+        require(address(beefy.want()) == address(lpToken), "BPS: Invalid beefy vault asset.");
+
+        // approve assets to use in pearl router
+        token0.approve(address(pearlRouter), type(uint256).max);
+        token1.approve(address(pearlRouter), type(uint256).max);
+
+        // approve lp token for pearlRouter and beefy vault
+        lpToken.approve(address(pearlRouter), type(uint256).max);
+        lpToken.approve(address(beefy), type(uint256).max);
+    }
+
+    // TODO: parameterize (address from, address to)
+    function _getInLPRatio() internal view returns (uint256, uint256) {
```

Comment:
I don't think this function is needed. See the comment on `_investIntoBeefy` below.